megamek/src/megamek/client/ui/swing/lobby/LobbyActions.java

```diff
@@ -0,0 +1,1312 @@
+        }
+        Set<Entity> updateCandidates = new HashSet<>();
+        boolean goProne = info.equals("PRONE");
+        boolean goHullDown = info.equals("HULLDOWN");
+        boolean stand = !goProne && !goHullDown;
+        for (Entity entity: entities) {
+            if ((goProne && !entity.isProne()) || (goHullDown && !entity.isHullDown())
+                    || (stand && (entity.isProne() || entity.isHullDown()))) {
+                entity.setProne(goProne);
+                entity.setHullDown(goHullDown);
+                updateCandidates.add(entity);
+            }
+        }
+        sendUpdates(updateCandidates);
+    }
+    
+    /**
+     * Attaches the given force as a subforce to the given new parent. 
+     * Does NOT work for newParentId == NO_FORCE. Use promoteForce to do this.
+     * Does not allow attaching a force to one of its own subforces.
+     */
+    void forceAttach(int forceId, int newParentId) {
+        Forces forces = game().getForces();
+        if (!forces.contains(forceId) || !forces.contains(newParentId)
+                || (forceId == newParentId)) {
+            return;
+        }
+        
+        Force force = forces.getForce(forceId);
+        Force newParent = forces.getForce(newParentId);
+        List<Force> subForces = forces.getFullSubForces(force);
+        IPlayer owner = forces.getOwner(force);
+        IPlayer newParentOwner = forces.getOwner(newParent);
+            
+        if (owner.isEnemyOf(newParentOwner)) {
+            LobbyErrors.showOnlyTeam(frame());;
+            return;
+        }
+        if (subForces.contains(newParent)) {
+            LobbyErrors.showForceNoAttachSubForce(frame());
+            return;
+        }
+        if (!isEditable(force)) {
+            LobbyErrors.showCannotConfigEnemies(frame());
+            return;
+        }
+        var forceList = new ArrayList<Force>(List.of(force));
+        client().sendForceParent(forceList, newParentId);
+    }
+    
+
+    /**
+     * Makes the given forces top-level, detaching them from any former parents. 
+     */
+    void forcePromote(Collection<Integer> forceIds) {
+        var forces = game().getForces();
+        if (forceIds.stream().anyMatch(id -> !forces.contains(id))) {
+            return;
+        }
+        Set<Force> forceList = forceIds.stream().map(id -> forces.getForce(id)).collect(toSet());
+        if (!areForcesEditable(forceList)) {
+            LobbyErrors.showCannotConfigEnemies(frame());
+            return;
+        }
+        client().sendForceParent(forceList, Force.NO_FORCE);
+    }
+
+    /** Shows the dialog which allows adding pre-existing damage to units. */
+    void configureDamage(Collection<Entity> entities) {
+        if (entities.size() != 1) {
+            LobbyErrors.showSingleUnit(frame(), "assign damage");
+            return;
+        }
+        if (!validateUpdate(entities)) {
+            return;
+        }
+        Entity entity = CollectionUtil.randomElement(entities);

```

Comment:
Here and at other places, I wanted to receive Collections instead of lists (as discussed above), especially as the order of the selected entities shouldnt play any role. When the collection has only 1 element its still bothersome to get that element, therefore I used that method. Sometimes I used it with multiple entities to find some, e.g. to find one owner. I think I'll rename it to anyOneElement or something like that as it really has nothing to do with random.